src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/FileExtensionsAttributeAdapter.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using Microsoft.Extensions.Localization;
+
+namespace Microsoft.AspNetCore.Mvc.DataAnnotations.Internal
+{
+    public class FileExtensionsAttributeAdapter : AttributeAdapterBase<FileExtensionsAttribute>
+    {
+        public FileExtensionsAttributeAdapter(FileExtensionsAttribute attribute, IStringLocalizer stringLocalizer)
+            : base(attribute, stringLocalizer)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void AddValidation(ClientModelValidationContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            MergeAttribute(context.Attributes, "data-val", "true");
+            MergeAttribute(context.Attributes, "data-val-fileextensions", GetErrorMessage(context));
+            MergeAttribute(context.Attributes, "data-val-fileextensions-extensions", Attribute.Extensions);
```

Comment:
@dougbu any suggestions on how the 'extensions' should be handled in terms of case sensitiveness? do we want it to be case-insensitive always? (in view of an app being hosted in Windows OSes or Linux/Mac)

Reply:
How is case-sensitivity relevant here?

Reply:
If a property were decorated like `FileExtensions(Extensions = "JPEG, Png")`, should the jquery validation exactly match the file extensions with case? if we always want it to be case insensitive, should the adapter here lower case upfront so the generated html has the data attributes with lower case values?

Reply:
Ah. This isn't about case-sensitivity per se; it's about normalizing the `Extensions` as the attribute does for use in the error message.

Yes, do that normalization here. That should simplify the jQuery Validation changes.